### DISCUSSION TITLE
How to remove false warnings from vsCode

### DISCUSSION THREAD

**Discussion by gcraill** [zendtech/IbmiToolkit#190] [Q&A]
Hi,

We are using vsCode with the Seiden Group Community+ edition of PHP (seiden_stable_81) and when we have toolkit commands in the php code we get errors displaying on vscode. These are cosmetic as deploying the code works fine but I would like to be able to remove the warnings.

Does anyone know how to configure vscode to parse and accept the toolkit commands?

![image](https://github.com/zendtech/IbmiToolkit/assets/30224542/9745690f-5736-4316-a407-79a4bf3ceebe)

I added the extra comments to give a bit of context but we in this case just use that one line of code.

Cheers
Greg

**Comment by NattyNarwhal** [comment]
Should be a matter of making your editor aware of what's in the include path when code is included. I don't use VS Code, so I'm not sure how to do it there specifically.

**Comment by brandonp42** [comment]
@gcraill how is your environment set up? Ie are you editing your PHP code locally on your PC and then pushing to the server or are you editing directly on the server via SSH?

Also are you using Composer to install the toolkit?

**Reply by gcraill** [reply]
Hi Brandon,
We are not using composer.
We are editing on a local copy of the azure repo on our laptop and then deploying (to test instance) via the vscode deploy option.  (we connect to an IBM i lpar from vscode and select the target folder as the deploy destination). From that I am wondering if we can load the toolkit folders locally and reference them on the laptop?
For the prod instances we deploy via an azure pipeline.
Cheers
Greg

**Comment by alanseiden** [comment]
@gcraill Would you consider loading the toolkit locally (on your PC) by using Composer in your project? Then VS Code will know how to find the toolkit locally.

**Reply by zkarj735** [reply]
Greg is deferring to me on this re Composer (we are teammates).

I see a Composer plugin for VSCode, however in the configuration section it mentions the need for a PHP executable, which is where we come unstuck. Our PCs are very locked down meaning we cannot install much of anything.

Already we get regular errors pop up on VSCode launch saying it could not find the PHP executable path, because we do not (and cannot) have one, yet it seems to happily auto-complete and syntax highlight *except* for the Toolkit methods.

I had previously tried manually downloading a copy of the Toolkit files and setting `@ext:bmewburn.vscode-intelephense-client Paths` to point to that, however this was ineffective.

**Reply by zkarj735** [reply]
![CleanShot 2023-08-29 at 14 17 59@2x](https://github.com/zendtech/IbmiToolkit/assets/62422779/e0953def-4621-4760-935c-ef33640b0500)
